### PR TITLE
Add chromatic harps

### PR DESCRIPTION
**Pull request by js-jslog**
Adding chromatic harps is going to require quite an overhaul because there is a baked in assumption at the moment that only one harp face will be presetable at a time.

### CODE REVIEW

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-fullmatrix-to-columnbounds/reduce-fullmatrix-to-columnbounds.ts [hunk 1]
```diff
@@ -10,9 +10,11 @@ import type { ColumnBounds } from '../../../../types'
 
 export const reduceFullMatrixToColumnBounds = (
   prevColumnBounds: ColumnBounds,
-  fullMatrix: HarpFaceMatrix<unknown>
+  fullMatrix: HarpFaceMatrices<unknown>
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-fullmatrix-to-columnbounds/reduce-fullmatrix-to-columnbounds.ts:13]: As well as having a need to rename many of the global state props recognised in https://github.com/js-jslog/harpguru/issues/120, we now need to reconsider everything which is named a matrix but is actually one of these potential paired matrices.

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fulldegreematrix/reduce-harpstrata-to-fulldegreematrix.ts [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
 export const reduceHarpStrataToFullDegreeMatrix = (
-  prevDegreeMatrix: HarpFaceMatrix<Degree>,
+  prevDegreeMatrix: HarpFaceMatrices<Degree>,
   harpStrata: HarpStrata
-): HarpFaceMatrix<Degree> => {
+): HarpFaceMatrices<Degree> => {
   const { degreeMatrix: nextDegreeMatrix } = harpStrata
-  if (doSparceIdedObjectMatricesMatch(prevDegreeMatrix, nextDegreeMatrix))
+  if (
+    isMatchHarpFaceMatrices(
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fulldegreematrix/reduce-harpstrata-to-fulldegreematrix.ts]: Should maybe consider `HarpFaceMatrices` as a more accurate name where variables which are of that type or associated with that type have less helpful names.

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullinteractionmatrix/reduce-harpstrata-to-fullinteractionmatrix.test.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ test('the previous fullInteractionMatrix object is returned if it matches the on
   const {
     apparatus: { interactionMatrix },
   } = harpStrata
-  const prevInteractionMatrix = [...interactionMatrix]
+  const prevInteractionMatrix = { ...interactionMatrix }
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullinteractionmatrix/reduce-harpstrata-to-fullinteractionmatrix.test.ts:12]: `interactionMatrix` is another example of a property which needs changing and will have wide ramifications. `interactionMatrices` at the very least, but probably also `interactionHarpFaceMatrices` should be considered.

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullinteractionmatrix/reduce-harpstrata-to-fullinteractionmatrix.ts [hunk 4]
```diff
@@ -1,17 +1,19 @@
 import type { HarpStrata } from 'harpstrata'
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullinteractionmatrix/reduce-harpstrata-to-fullinteractionmatrix.ts:1]: Even this filename and filenames like it are going to have to change when the property names are changed.

A tip to follow will be to search for filenames with the properties in at the same time as the properties are updated.

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullpitchmatrix/reduce-harpstrata-to-fullpitchmatrix.test.ts [hunk 5]
```diff
@@ -37,3 +41,6 @@ test('the next fullPitchMatrix object is returned if it is different from the pr
   expect(nextPitchMatrix).not.toStrictEqual(prevPitchMatrix)
   expect(nextPitchMatrix).toBe(harpStrata.pitchMatrix)
 })
+
+// TODO: Add test for when the the second matrix isn't empty. All the source data has empty second matrices at the moment
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullpitchmatrix/reduce-harpstrata-to-fullpitchmatrix.test.ts]: A couple of TODO's here. The second echo's the sentiment of a couple of code review comments already made here.

File: packages/harpguru-core/src/components/harp-cell/harp-cell.test.tsx [hunk 6]
```diff
@@ -18,7 +18,8 @@ test('A component is rendered with the Degree or Pitch value in its text view de
   mockUseGlobal.mockImplementation(buildMockUseGlobalImplementation({}))
   const harpCellProps = {
     yxCoord: [3, 0] as [3, 0],
-  }
+    harpfaceIndex: 'harpface1',
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/harp-cell/harp-cell.test.tsx:21]: Are we sure about the `harpface1` & `harpface2` strings..? I suppose they are more meaningful than just a numerical index. If the parameter is well named though then perhaps it doesn't need to have a meaningful string..

Comment (js-jslog) [reply packages/harpguru-core/src/components/harp-cell/harp-cell.test.tsx:21]: Perhaps leaving these properties as they are is a good idea for now because it will allow an easier search for all contexts where a chromatic oriented test could be added.

File: packages/harpguru-core/src/components/harp-cell/hooks/use-position-analysis/use-position-analysis.ts [hunk 7]
```diff
@@ -11,11 +12,17 @@ export type PositionFacts = {
...
 }
 
-export const usePositionAnalysis = (yxCoord: YXCoord): PositionFacts => {
-  const [degreeMatrix] = useGlobal('activeDegreeMatrix')
-  const [pitchMatrix] = useGlobal('activePitchMatrix')
+export const usePositionAnalysis = (
+  yxCoord: YXCoord,
+  harpfaceIndex: 'harpface1' | 'harpface2'
+): PositionFacts => {
+  const [degreeMatrices] = useGlobal('activeDegreeMatrix')
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/harp-cell/hooks/use-position-analysis/use-position-analysis.ts:19]: Classic example of why we need to update the global state property names. This should be `fullDegreeMatrices` or `fullDegreeHarpFaceMatrices` or even `degreeFullHarpFaceMatrices`.

File: packages/harpguru-core/src/components/harp-face/hooks/use-styles.ts [hunk 8]
```diff
@@ -2,22 +2,30 @@ import { useGlobal } from 'reactn'
...
 
   const [layoutFacts] = useGlobal('layoutFacts')
-  const { harpfaceRows, harpfaceColumns } = layoutFacts
-
-  const octaveColumnGroups = useOctaveColumnGroups()
+  const {
+    [harpfaceIndex === 'harpface1' ? 0 : 1]: { harpfaceRows, harpfaceColumns },
+  } = layoutFacts
+
+  // TODO: Should we perhaps just make useOctaveColumnGroups always use
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/harp-face/hooks/use-styles.ts]: Yes, definitely for now. It's a clear indication of design intent. I have no usecase for fragmenting by the second harpfaces octave boundaries.

File: packages/harpguru-core/src/components/harp-faces/harp-faces.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      ...StyleSheet.absoluteFillObject,
+      flexDirection: 'column',
+      alignItems: 'center',
+      justifyContent: 'center',
+    },
+    gutter: {
+      height: fragmentGutter,
+    },
+  })
+  // TODO: Have this as part of the layout facts
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/harp-faces/harp-faces.tsx]: The problem with this is that at the moment we are only spending a rerender of this component when the interaction matrix is updated, but the LayoutFacts might be updated more regularly than that and produce unnecessary renders.

There is also the problem that having a property in global state will not give us the same typeguarding assurances that actually using the typeguard here will. There is more potential for misalignment.

File: packages/harpguru-core/src/components/harp-faces/harp-faces.tsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    )
+  }
+  return (
+    <View style={stack}>
+      <HarpFace harpfaceIndex={'harpface1'} />
+    </View>
+  )
+}
+
+// TODO: memoise this component to only return a new one if the number of harpfaces changes
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/harp-faces/harp-faces.tsx]: After some experimentation it doesn't look like there is much to be gained here.

File: packages/harpguru-core/src/components/menu-of-tunings/hooks/use-zoom-items/use-zoom-items.tsx [hunk 11]
```diff
@@ -29,7 +29,9 @@ export const useZoomItems = (
       value={getZoomText(id)}
       isSelected={id === activeZoomId}
       itemTapHandler={itemTapHandler}
-      degreeMatrix={activeDegreeMatrix}
+      // TODO: This really doesn't need to be either of the
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/menu-of-tunings/hooks/use-zoom-items/use-zoom-items.tsx:32]: TODO here. I'm going to have to double check what this is actually getting at.

File: packages/harpguru-core/src/global.d.ts [hunk 12]
```diff
@@ -33,17 +33,19 @@ declare module 'reactn/default' {
...
     readonly rootPitchId: PitchIds
     readonly harpKeyId: PitchIds
-    readonly viewableInteractionMatrix: HarpFaceMatrix<Interaction>
-    readonly viewableDegreeMatrix: HarpFaceMatrix<Degree>
-    readonly viewablePitchMatrix: HarpFaceMatrix<Pitch>
-    readonly layoutFacts: LayoutFacts
+    readonly viewableInteractionMatrix: HarpFaceMatrices<Interaction>
+    readonly viewableDegreeMatrix: HarpFaceMatrices<Degree>
+    readonly viewablePitchMatrix: HarpFaceMatrices<Pitch>
+    // TODO: It would be nice to have a type which is more in keeping with the
```
Comment (js-jslog) [inline packages/harpguru-core/src/global.d.ts]: TODO here. Yes this would be a good idea. `HarpFacePossiblePair` perhaps.

File: packages/harpguru-core/src/hooks/use-octave-column-groups/use-octave-column-groups.test.ts [hunk 13]
```diff
@@ -1,6 +1,6 @@
 import { useGlobal } from 'reactn'
```
Comment (js-jslog) [inline packages/harpguru-core/src/hooks/use-octave-column-groups/use-octave-column-groups.test.ts]: It would definitely be worth considering keeping this function clear of the override, and having  it always return the octave column groups. It could then be up to the caller to flatten it if it were relevant.

Comment (js-jslog) [reply packages/harpguru-core/src/hooks/use-octave-column-groups/use-octave-column-groups.test.ts]: `useOctaveColumnGroups` & `useFragmentColumnGroups`

File: packages/harpguru-core/src/hooks/use-sizes/use-sizes.ts [hunk 14]
```diff
@@ -45,11 +47,14 @@ const relativeLabelIconSize = 7
```
Comment (js-jslog) [inline packages/harpguru-core/src/hooks/use-sizes/use-sizes.ts]: This file deserves a lot of isolation an util functions at this point.

File: packages/harpguru-core/src/utils/determine-matrix-dimensions/determine-matrix-dimensions.ts [hunk 15]
```diff
@@ -7,9 +7,10 @@ export const determineMatrixDimensions = (
   matrix: ReadonlyArray<ReadonlyArray<unknown>>
 ): MatrixDimensions => {
   const { length: rows } = matrix
-  const {
-    [0]: { length: columns },
-  } = matrix
+  const columns = (() => {
```
Comment (js-jslog) [inline packages/harpguru-core/src/utils/determine-matrix-dimensions/determine-matrix-dimensions.ts:10]: This is necessitated by some basic assumption about the empty harpface2 matrix. I should review whether this is necessary or the most semantic solution.

File: packages/harpguru-core/src/utils/index.ts [hunk 16]
```diff
@@ -5,8 +5,11 @@ export { getColors } from './get-colors'
 export { determineZoomId } from './determine-zoom-id'
 export { determineMatrixDimensions } from './determine-matrix-dimensions'
 export { determineNextColumnBounds } from './determine-next-column-bounds'
+export { extractHarpFaceMatrix } from './extract-harpfacematrix'
 export { isMatchActiveIds } from './ismatch-activeids'
 export { isMatchColumnBounds } from './ismatch-columnbounds'
+export { isMatchHighOrderTuples } from './ismatch-highordertuples'
```
Comment (js-jslog) [inline packages/harpguru-core/src/utils/index.ts]: Is this still used? If it's only used for the layoutFacts then it can be removed if we get a bit more type parity between the layout facts and the harpfacematrices.

File: packages/harpguru-core/src/utils/ismatch-harpfacematrices/ismatch-harpfacematrices.ts [hunk 17]
```diff
@@ -0,0 +1,25 @@
+import { isChromaticHarpFace } from 'harpparts'
+import type { HarpFaceMatrices, HarpFaceMatrix } from 'harpparts'
+
+// TODO: This could use a test file
```
Comment (js-jslog) [inline packages/harpguru-core/src/utils/ismatch-harpfacematrices/ismatch-harpfacematrices.ts]: It could, but perhaps hold off until we've made a decision about the layoutfacts type.

File: packages/harpguru-core/src/utils/ismatch-highordertuples/ismatch-highordertuples.ts [hunk 18]
```diff
@@ -0,0 +1,12 @@
+// TODO: This could use a test file
```
Comment (js-jslog) [inline packages/harpguru-core/src/utils/ismatch-highordertuples/ismatch-highordertuples.ts]: It could, but perhaps hold off until we've made a decision about the layoutfacts type.

File: packages/harpguru-core/src/utils/map-row-to-remove-by-sibling-interaction/map-row-to-remove-by-sibling-interaction.test.ts [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+  )
+  const expectedMappedDegreeMatrix = [
+    [undefined, undefined, undefined, undefined],
+    [undefined, undefined, THIRD, undefined],
+  ]
+
+  expect(mappedDegreeMatrix).toStrictEqual(expectedMappedDegreeMatrix)
+})
+
+// TODO: There are more tests to write here. At least around the exception being thrown.
```
Comment (js-jslog) [inline packages/harpguru-core/src/utils/map-row-to-remove-by-sibling-interaction/map-row-to-remove-by-sibling-interaction.test.ts]: Better have a review.

File: packages/harpguru-core/src/utils/reduce-fullmatrix-to-viewablematrix/reduce-fullmatrix-to-viewablematrix.ts [hunk 20]
```diff
@@ -1,30 +1,103 @@
...
-  if (columnBounds === 'FIT') {
-    if (doSparceIdedObjectMatricesMatch(prevViewableMatrix, fullMatrix))
-      return prevViewableMatrix
-    return fullMatrix
-  }
+): HarpFaceMatrices<T> => {
+  const nextViewableMatrix = (() => {
+    if (isChromaticHarpFace(fullMatrix)) {
+      if (!isChromaticHarpFace(fullInteractionMatrix)) throw Error(errorMessage)
+      // TODO: This is rubbish. Can I not find a way to just remove the Blow & Draw
```
Comment (js-jslog) [inline packages/harpguru-core/src/utils/reduce-fullmatrix-to-viewablematrix/reduce-fullmatrix-to-viewablematrix.ts]: I have tried hard here already, but I really don't like it in this state. Perhaps it's worth another try to find a way to filter an enum list.

File: packages/harpparts/src/apparatus/utils/build-apparatus/index.ts [hunk 21]
```diff
@@ -1,17 +1,38 @@
...
 import { getTuning } from '../../../access-parts'
 
 export const buildApparatus = (
   tuningId: TuningIds,
   valvingId: ValvingIds
 ): Apparatus => {
-  const tuning = getTuning(tuningId)
+  const { reedArrays } = getTuning(tuningId)
+  const [reedArray1, reedArray2] = reedArrays
+  // TODO: the conditional on the possibility of reedArray2 being empty
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/build-apparatus/index.ts]: There's definitely some thought to put in here. This comment is enough for this code review. If I don't find a solution then a more detailed one will be due.

File: packages/harpparts/src/tuning/types/tuning-types.ts [hunk 22]
```diff
@@ -62,16 +63,19 @@ export type Tuning = {
   readonly id: TuningIds
   readonly shortName?: string
   readonly category: TuningCategories
-  readonly reedArray: ReedArray
+  readonly reedArrays: readonly [ReedArray, ReedArray]
```
Comment (js-jslog) [inline packages/harpparts/src/tuning/types/tuning-types.ts]: It might be beneficial to have some parity between the possibility of multiple reed arrays, and the possibility of multiple harpfaces. Even though they are in different projects I'm sure that would make things clearer.

Comment (js-jslog) [reply packages/harpparts/src/tuning/types/tuning-types.ts]: They're not even in different projects are they, it's just that we mostly use the harpfacematrices in the harpguru-core project, but they are defined here.

File: packages/harpparts/src/types/types.ts [hunk 23]
```diff
@@ -12,3 +12,13 @@ export type HalfstepIndex = number
 
 export type HarpFaceRow<T> = ReadonlyArray<T | undefined>
 export type HarpFaceMatrix<T> = ReadonlyArray<HarpFaceRow<T>>
+type HarpFaceMatricesDiatonic<T> = {
```
Comment (js-jslog) [inline packages/harpparts/src/types/types.ts]: This is the file where we perhaps need to consider the creation of a `PossiblePair` type, which can either be of HarpFaceMatrix, or of ReedArray or other types..

File: packages/harpstrata/src/get-harp-strata/get-harp-strata.ts [hunk 24]
```diff
@@ -9,26 +14,38 @@ export const getHarpStrata = (props: HarpStrataProps): HarpStrata => {
...
 
   const apparatus = buildApparatus(tuningId, valvingId)
+
   const { rootOffset } = getPozition(pozitionId)
   const pitch = getPitch(harpKeyId)
+  const { id: pitchId } = pitch
 
   const { halfstepIndexMatrix } = apparatus
-  const { id: pitchId } = pitch
+  const degreeMatrices = {
```
Comment (js-jslog) [inline packages/harpstrata/src/get-harp-strata/get-harp-strata.ts]: This could all be much nicer with a higher order function. That higher order function might require that the parameters that the function it will call takes be a single property so that type checking can be applied to it. Otherwise we'll have to give it a non-specific number of arguments to pass on to the function `getDegreeMatrix` in this case.

File: packages/harpstrata/src/get-harp-strata/utils/get-active-ids-pair/get-active-ids-pair.ts [hunk 25]
```diff
@@ -7,8 +7,14 @@ import {
 } from '../get-counterpart-active-ids'
 import type { IsActiveProps, ActiveIdsPair } from '../../../types'
 
+// TODO: This function definitely need to deal with both of the harpface matrices
```
Comment (js-jslog) [inline packages/harpstrata/src/get-harp-strata/utils/get-active-ids-pair/get-active-ids-pair.ts]: Big TODO here I think. There is the possibility that we will not get a complete list of active id's to convert or persist in to the next harpstrata unless we observe both of the harpfaces. I think that having a good think about what the type of the `harpface2` property should be will be key here and in other places to make things as slick as possible.

File: packages/harpparts/src/access-parts/constants/tunings.ts [hunk 26]
```diff
@@ -1,4 +1,4 @@
-import { TuningIds } from '../../tuning'
+import { SIXTEEN_HOLE_CHROMATIC, TuningIds } from '../../tuning'
```
Comment (js-jslog) [inline packages/harpparts/src/access-parts/constants/tunings.ts]: Don't forget to add some more chromatics too, including the diminished one which was requested.

Comment (js-jslog) [reply packages/harpparts/src/access-parts/constants/tunings.ts]: I think that was specfically a seydel nonslide one. Check the email on it.

File: packages/harpguru-core/src/global.d.ts (hunk 12 above)
Comment (js-jslog) [reply packages/harpguru-core/src/global.d.ts]: Covered off in https://github.com/js-jslog/harpguru/pull/130. As acknowledged in https://github.com/js-jslog/harpguru/issues/131#issuecomment-981960861, the name `HarpFaceFacts` could be a bit better, but the improvements still stand with that name so I'll close this discussion.

File: packages/harpguru-core/src/utils/index.ts (hunk 16 above)
Comment (js-jslog) [reply packages/harpguru-core/src/utils/index.ts]: Removed in https://github.com/js-jslog/harpguru/pull/130

File: packages/harpguru-core/src/utils/ismatch-harpfacematrices/ismatch-harpfacematrices.ts (hunk 17 above)
Comment (js-jslog) [reply packages/harpguru-core/src/utils/ismatch-harpfacematrices/ismatch-harpfacematrices.ts]: Covered off in https://github.com/js-jslog/harpguru/pull/130. It turned out that this had a major bug in it which didn't come out until the test was written.

File: packages/harpguru-core/src/utils/ismatch-highordertuples/ismatch-highordertuples.ts (hunk 18 above)
Comment (js-jslog) [reply packages/harpguru-core/src/utils/ismatch-highordertuples/ismatch-highordertuples.ts]: Removed in https://github.com/js-jslog/harpguru/pull/130.

File: packages/harpparts/src/apparatus/utils/build-apparatus/index.ts (hunk 21 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/build-apparatus/index.ts]: The complexity here has been dealt with as part of #130.

File: packages/harpparts/src/tuning/types/tuning-types.ts (hunk 22 above)
Comment (js-jslog) [reply packages/harpparts/src/tuning/types/tuning-types.ts]: Dealt with in #130 .

File: packages/harpparts/src/types/types.ts (hunk 23 above)
Comment (js-jslog) [reply packages/harpparts/src/types/types.ts]: Dealt with in #130.

File: packages/harpstrata/src/get-harp-strata/get-harp-strata.ts (hunk 24 above)
Comment (js-jslog) [reply packages/harpstrata/src/get-harp-strata/get-harp-strata.ts]: This has been handled in #130, but I'm going to leave this comment open as a way to remind that additional tests are required anywhere where the `mapHarpFaceFacts` function was used, to account for the possibility that a chromatic config could pass through the function.

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullpitchmatrix/reduce-harpstrata-to-fullpitchmatrix.test.ts (hunk 5 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullpitchmatrix/reduce-harpstrata-to-fullpitchmatrix.test.ts]: Done.

File: packages/harpguru-core/src/components/harp-cell/harp-cell.test.tsx (hunk 6 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/harp-cell/harp-cell.test.tsx:21]: I could define a couple of constants somewhere, but having the type definition makes that unnecessary really.

File: packages/harpguru-core/src/components/harp-face/hooks/use-styles.ts (hunk 8 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/harp-face/hooks/use-styles.ts]: Done

File: packages/harpguru-core/src/components/harp-faces/harp-faces.tsx (hunk 9 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/harp-faces/harp-faces.tsx]: This has been mentioned in the following issues:

- https://github.com/js-jslog/harpguru/issues/134
- https://github.com/js-jslog/harpguru/issues/135

File: packages/harpguru-core/src/components/harp-faces/harp-faces.tsx (hunk 10 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/harp-faces/harp-faces.tsx]: That's not true, but issues have now been raised - see https://github.com/js-jslog/harpguru/pull/129#discussion_r761850991

File: packages/harpguru-core/src/hooks/use-octave-column-groups/use-octave-column-groups.test.ts (hunk 13 above)
Comment (js-jslog) [reply packages/harpguru-core/src/hooks/use-octave-column-groups/use-octave-column-groups.test.ts]: Done. The size scheme is no longer aware of the column groups so there is no need for 2 hooks.

File: packages/harpguru-core/src/utils/determine-matrix-dimensions/determine-matrix-dimensions.ts (hunk 15 above)
Comment (js-jslog) [reply packages/harpguru-core/src/utils/determine-matrix-dimensions/determine-matrix-dimensions.ts:10]: It's fine. Clear enough and functional.

File: packages/harpstrata/src/get-harp-strata/get-harp-strata.ts (hunk 24 above)
Comment (js-jslog) [reply packages/harpstrata/src/get-harp-strata/get-harp-strata.ts]: The only places which don't have additional tests now are `getHarpStrata` & `buildApparatus`.

Comment (js-jslog) [reply packages/harpstrata/src/get-harp-strata/get-harp-strata.ts]: Raised as techdebt: https://github.com/js-jslog/harpguru/issues/131#issuecomment-985442182

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-fullmatrix-to-columnbounds/reduce-fullmatrix-to-columnbounds.ts (hunk 1 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-fullmatrix-to-columnbounds/reduce-fullmatrix-to-columnbounds.ts:13]: Raised as techdebt in https://github.com/js-jslog/harpguru/issues/131#issuecomment-985444203

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fulldegreematrix/reduce-harpstrata-to-fulldegreematrix.ts (hunk 2 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fulldegreematrix/reduce-harpstrata-to-fulldegreematrix.ts]: Raised as techdebt in https://github.com/js-jslog/harpguru/issues/131#issuecomment-985444203

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullinteractionmatrix/reduce-harpstrata-to-fullinteractionmatrix.test.ts (hunk 3 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullinteractionmatrix/reduce-harpstrata-to-fullinteractionmatrix.test.ts:12]: Raised as techdebt in https://github.com/js-jslog/harpguru/issues/131#issuecomment-985444203

File: packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullinteractionmatrix/reduce-harpstrata-to-fullinteractionmatrix.ts (hunk 4 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/callback-on-sourceglobalprops/utils/reduce-harpstrata-to-fullinteractionmatrix/reduce-harpstrata-to-fullinteractionmatrix.ts:1]: Raised as techdebt in https://github.com/js-jslog/harpguru/issues/131#issuecomment-985444203

File: packages/harpguru-core/src/components/harp-cell/hooks/use-position-analysis/use-position-analysis.ts (hunk 7 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/harp-cell/hooks/use-position-analysis/use-position-analysis.ts:19]: Raised as techdebt in https://github.com/js-jslog/harpguru/issues/131#issuecomment-985444203

File: packages/harpguru-core/src/hooks/use-sizes/use-sizes.ts (hunk 14 above)
Comment (js-jslog) [reply packages/harpguru-core/src/hooks/use-sizes/use-sizes.ts]: This will be handled by https://github.com/js-jslog/harpguru/issues/134

File: packages/harpguru-core/src/utils/map-row-to-remove-by-sibling-interaction/map-row-to-remove-by-sibling-interaction.test.ts (hunk 19 above)
Comment (js-jslog) [reply packages/harpguru-core/src/utils/map-row-to-remove-by-sibling-interaction/map-row-to-remove-by-sibling-interaction.test.ts]: Done

File: packages/harpstrata/src/get-harp-strata/utils/get-active-ids-pair/get-active-ids-pair.ts (hunk 25 above)
Comment (js-jslog) [reply packages/harpstrata/src/get-harp-strata/utils/get-active-ids-pair/get-active-ids-pair.ts]: On closer inspection this isn't as much of a problem as it first seemed. It could use fixing, but it will not represent a significant risk the way things are setup at the moment. See raised issue: https://github.com/js-jslog/harpguru/issues/136

File: packages/harpguru-core/src/components/menu-of-tunings/hooks/use-zoom-items/use-zoom-items.tsx (hunk 11 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/menu-of-tunings/hooks/use-zoom-items/use-zoom-items.tsx:32]: Raised to be considered as part of #134 & #135

File: packages/harpguru-core/src/utils/reduce-fullmatrix-to-viewablematrix/reduce-fullmatrix-to-viewablematrix.ts (hunk 20 above)
Comment (js-jslog) [reply packages/harpguru-core/src/utils/reduce-fullmatrix-to-viewablematrix/reduce-fullmatrix-to-viewablematrix.ts]: Raised as techdebt: https://github.com/js-jslog/harpguru/issues/131#issuecomment-985588026

File: packages/harpguru-core/src/hooks/use-octave-column-groups/use-octave-column-groups.test.ts [hunk 27]
```diff
@@ -1,14 +1,18 @@
...
 
-import { buildMockUseGlobalImplementation } from '../../../../test-resources'
+import { buildMockUseGlobalImplementation } from '../../test-resources'
 
 import { useOctaveColumnGroups } from './use-octave-column-groups'
```
Comment (js-jslog) [inline packages/harpguru-core/src/hooks/use-octave-column-groups/use-octave-column-groups.test.ts]: We have moved the `useOctaveColumnGroups` hook to the project root hooks folder, but it turns out that that wasn't actually necessary. Move it back to the `harp-face` folder.

Comment (js-jslog) [reply packages/harpguru-core/src/hooks/use-octave-column-groups/use-octave-column-groups.test.ts]: Done

File: packages/harpparts/src/access-parts/constants/tunings.ts (hunk 26 above)
Comment (js-jslog) [reply packages/harpparts/src/access-parts/constants/tunings.ts]: Added as techdebt: https://github.com/js-jslog/harpguru/issues/131#issuecomment-985646857

**Review by js-jslog — COMMENTED** [review]
LGTM
